internal/controllers/machine/machine_controller_phases.go

```diff
@@ -372,3 +362,31 @@ func (r *Reconciler) reconcileCertificateExpiry(ctx context.Context, _ *clusterv
 
 	return ctrl.Result{}, nil
 }
+
+// removeOutdatedOwnerRefs will remove any MachineSet or ControlPlane ownerReferences from Infrastructure Machine and
+// Bootstrap objects referred to by the machine. These ownerReferences are added on  creation as the Machine does not exist
+// when the objects are initially created.
+func removeOutdatedOwnerRefs(cluster *clusterv1.Cluster, m *clusterv1.Machine, obj *unstructured.Unstructured) {
+	cpGVK := getControlPlaneGVKForMachine(cluster, m)
+	for _, owner := range obj.GetOwnerReferences() {
+		if owner.APIVersion == clusterv1.GroupVersion.String() && owner.Kind == "MachineSet" ||
```

Comment:
Let's use `()` for each part of the `||` not sure if the default evaluation groups the conditions as we expect it to